### PR TITLE
Use Sequence instead of List for DataModelApply.view

### PR DESCRIPTION
**Pull request by erlendvollset**
- **Bump mypy**
- **Use Sequence instead of list in DataModelApply.views**

## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -51,6 +51,7 @@ shapely = { version = ">=1.7.0", optional = true }
 pyodide-http = { version = "^0.2.1", optional = true }
 graphlib-backport = { version = "^1.0.0", python = "<3.9" }
 PyYAML = { version = "^6.0", optional = true }
+mypy = "^1"
```
Comment (haakonvt) [inline pyproject.toml:54]: Should only be a dev deps, I think

Comment (erlendvollset) [reply pyproject.toml:54]: indeed

Comment (erlendvollset) [reply pyproject.toml]: fixed

**Review by haakonvt — APPROVED** [review]
LGTM

**Pull request merged**